### PR TITLE
test(changelog): Leaner, DRYer tests

### PR DESCRIPTION
**Pull request by BYK**

### CODE REVIEW

File: src/utils/__tests__/changelog.test.ts [hunk 1]
```diff
@@ -304,391 +305,314 @@ describe('generateChangesetFromGit', () => {
     >).mockReturnValue({ graphql: mockClient });
   });
 
-  it('should not fail on empty changeset', async () => {
-    mockGetChangesSince.mockResolvedValueOnce([]);
-    const changes = await generateChangesetFromGit(dummyGit, '1.0.0');
-    expect(changes).toBe('');
-  });
+  function setup(
+    commits: any[],
```
Comment (BYK) [inline src/utils/__tests__/changelog.test.ts:309]: Will improve typing here.

File: src/utils/__tests__/changelog.test.ts [hunk 2]
```diff
@@ -304,391 +281,346 @@ describe('generateChangesetFromGit', () => {
...
-      },
-    ]);
-    mockClient.mockResolvedValueOnce({
-      repository: {
-        M1: {
+        '1': {
           title: 'Better drivetrain',
           state: 'CLOSED',
         },
       },
```
Comment (iker-barriocanal) [inline src/utils/__tests__/changelog.test.ts:533]: What `body` does GitHub return on an empty milestone body? We're assuming the body to be trimmed, but does this mean it's not defined, or to be `''`? In a [past PR](https://github.com/getsentry/craft/pull/296#discussion_r715568597) this was discussed when `description` was `''`. Bodies, when empty, they are empty strings, why? Or, why milestones aren't?

Comment (BYK) [reply src/utils/__tests__/changelog.test.ts:533]: You just get a `null` in place of the milestone. The `description` comes as an empty string when it is not filled. I'll make the adjustment in code to clarify it.

**Review by iker-barriocanal — APPROVED** [review]

**Review by chadwhitacre — APPROVED** [review]
There we go. :)

**Pull request merged**